src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp

```diff
@@ -0,0 +1,750 @@
+
+    const auto ydot_slice = (i_ytilda + 1) * ydot <= y ? ydot : y % ydot;
+    const auto xdot_slice = (i_xtilda + 1) * xdot <= x ? xdot : x % xdot;
+
+    const auto gemm_m = c;
+    const auto gemm_n = n * htilda_slice * wtilda_slice;
+    const auto gemm_k = k * ydot_slice * xdot_slice;
+
+    return std::make_tuple(gemm_m, gemm_n, gemm_k);
+}
+
+// TODO: add fp16 and bfp16 by ConvHipImplicitGemmBwdDataV4R1Xdlops::GetWorkspaceSize(const
+// ConvolutionContext& ctx) const
+
+bool ConvHipImplicitGemmBwdDataV4R1Xdlops::IsApplicable(const ConvolutionContext& ctx) const
+{
+#if WORKAROUND_SWDEV_229277_227616_229195
+    if(!IsHccCompiler())
+        return false;
+#endif
+    bool is_applicable = true;
+
+    if(!ctx.direction.IsBackwardData())
+        return false;
+
+    if(!ctx.Is2d())
+        return false;
+
+    if(!ctx.IsFp32())
+        return false;
+
+    if(ctx.group_counts != 1)
+        return false;
+
+    if(!IsApplicableXdlops(ctx))
+        return false;
+
+    int gemm_m = 0;
+    int gemm_n = 0;
+
+    std::tie(gemm_m, gemm_n, std::ignore) = CalculateGemmSize(ctx, 0);
+
+    is_applicable = is_applicable && gemm_m % 32 == 0 && gemm_n % 32 == 0;
+
+    for(int gemm_id = 0; gemm_id < CalculateNumberOfGemm(ctx); ++gemm_id)
+    {
+        int gemm_k = 0;
+
+        std::tie(std::ignore, std::ignore, gemm_k) = CalculateGemmSize(ctx, gemm_id);
+
+        is_applicable = is_applicable && gemm_k % 4 == 0;
+    }
+
+    return is_applicable;
+}
+
+PerformanceImplicitGemmBwdDataV4R1Xdlops
+ConvHipImplicitGemmBwdDataV4R1Xdlops::GetPerformanceConfig(const ConvolutionContext& ctx) const
+{
+    return GetPerformanceConfigBase<PerformanceImplicitGemmBwdDataV4R1Xdlops>(ctx);
+}
+
+bool ConvHipImplicitGemmBwdDataV4R1Xdlops::IsValidPerformanceConfig(
+    const ConvolutionContext& ctx, const PerformanceImplicitGemmBwdDataV4R1Xdlops& c) const
+{
+    MIOPEN_LOG_I("");
+    return c.IsValidValue() && c.IsValid(ctx);
+}
+PerformanceImplicitGemmBwdDataV4R1Xdlops
+ConvHipImplicitGemmBwdDataV4R1Xdlops::Search(const ConvolutionContext& ctx) const
+{
+    // \todo add fp16 and bfp16 kernels
+    return GenericSearchBwd(*this, ctx);
+}
+
+int ConvHipImplicitGemmBwdDataV4R1Xdlops::RunAndMeasureSolution(miopen::Handle& profile_h,
+                                                                ConstData_t bot_buf,
+                                                                Data_t top_buf,
+                                                                ConstData_t wei_buf,
+                                                                ConstData_t bias_buf,
+                                                                const ConvolutionContext&,
+                                                                const ConvSolution& solution,
+                                                                float& elapsed_time) const
+{
+    assert(bias_buf == nullptr);
+    (void)bias_buf;
+
+#ifdef NDEBUG
```

Comment:
Why need to ```try```? Is the kernel compilation failing?

Reply:
I didn't meet compilation failing cases. I saw that the other solvers used this try catch to avoid program crash. Is it could be removed?

Reply:
Why it fails in compilation?

Reply:
@asroy Hi, Chao, I never meet any compilation fail. The "try" here is aim to avoid a compiler bug.

Reply:
Compilation may fail during auto-tuning due to subtle errors in the kernel or in the Solver. Or, because of previously uncaught errors in the compiler. This happens from time to time. If we do not catch, the whole tuning session will be terminated. That is why we use this pattern.